File: RailgunVB/Core/Preconditions/BotAdmin.vb
Imports Discord
Imports Discord.Commands
Imports Microsoft.Extensions.DependencyInjection
Imports RailgunVB.Core.Configuration

Namespace Core.Preconditions
    
    <AttributeUsage(AttributeTargets.Class Or AttributeTargets.Method)>
    Public NotInheritable Class BotAdmin
        Inherits PreconditionAttribute

        Public Overrides Async Function CheckPermissionsAsync(context As ICommandContext, command As CommandInfo, services As IServiceProvider) As Task(Of PreconditionResult)
            Dim config As MasterConfig = services.GetService(Of MasterConfig)
            Dim user As IGuildUser = Await context.Guild.GetUserAsync(context.User.Id)
                
            If user.Id = config.DiscordConfig.MasterAdminId OrElse 
               config.DiscordConfig.OtherAdmins.Contains(user.Id) 
                Return Await Task.FromResult(PreconditionResult.FromSuccess())
            Else 
                Return Await Task.FromResult(PreconditionResult.FromError(
                    "You must be a Railgun Administrator to use this command."))
            End If
        End Function
                                   
    End Class
    
End NameSpace

File: Railgun/Core/Preconditions/BotAdmin.cs
using System;
using System.Threading.Tasks;
using Discord.Commands;
using Microsoft.Extensions.DependencyInjection;
using Railgun.Core.Configuration;

namespace Railgun.Core.Preconditions
{
    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
    public class BotAdmin : PreconditionAttribute
    {
        public override async Task<PreconditionResult> CheckPermissionsAsync(ICommandContext context, CommandInfo command, IServiceProvider services) {
            var config = services.GetService<MasterConfig>();
            var user = await context.Guild.GetUserAsync(context.User.Id);

            if (user.Id == config.DiscordConfig.MasterAdminId || config.DiscordConfig.OtherAdmins.Contains(user.Id))
                return await Task.FromResult(PreconditionResult.FromSuccess());
            
            return await Task.FromResult(PreconditionResult.FromError("You must be a Railgun Administrator to use this command."));
        }
    }
}